src/core/mount.c

```diff
@@ -1578,7 +1578,8 @@ static void mount_sigchld_event(Unit *u, pid_t pid, int code, int status) {
 
         if (IN_SET(m->state, MOUNT_REMOUNTING, MOUNT_REMOUNTING_SIGKILL, MOUNT_REMOUNTING_SIGTERM))
                 mount_set_reload_result(m, f);
-        else if (m->result == MOUNT_SUCCESS)
+        else if (m->result == MOUNT_SUCCESS && !IN_SET(m->state, MOUNT_MOUNTING, MOUNT_UNMOUNTING))
+                /* MOUNT_MOUNTING and MOUNT_UNMOUNTING states need to be patched, see below. */
```

Comment:
Rather than patching result later, patch them here.
Otherwise, `unit_log_process_exit()` logs the failure louder.

Reply:
So I thought about this, but umount(8) already logs about the failure, and it would be strange/possibly confusing if we patch the result silently. LOG_NOTICE is fine IMO?

Reply:
Indeed...